public/app/features/dashboard/state/PanelModel.ts

```diff
@@ -458,7 +458,8 @@ export class PanelModel implements DataConfigSource, IPanelModel {
 
     if (plugin.onPanelMigration) {
       if (version !== this.pluginVersion) {
-        this.options = plugin.onPanelMigration(this);
+        const newPanelOptions = plugin.onPanelMigration(this);
+        this.options = await newPanelOptions;
```

Comment:
Q: do we need to add a check to see if the result is a Promise? Or can you await none promise-based results?

Reply:
you can await synchronous things, that's one of the best bits about `await` 😍 https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#:~:text=The%20expression%20is%20resolved%20in%20the%20same%20way%20as%20Promise.resolve()%3A%20it%27s%20always%20converted%20to%20a%20native%20Promise%20and%20then%20awaited

Reply:
Super! That is great 🙌🏻